core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -222,10 +222,15 @@
     void checkCanShowColumns(SecurityContext context, CatalogSchemaTableName table);
 
     /**
-     * Filter the list of columns to those visible to the identity.
+     * Filter the list of columns to those visible to the identity when querying metadata.
```

Comment:
This needs a better explanation. What's the difference between "when querying metadata" vs "when querying the table"?

Reply:
filterColumns hides table metadata when running DESCRIBE or SHOW TABLE, as well as when querying the INFORMATION_SCHEMA tables. However, when actually selecting data from the table it doesn't hide the corresponding columns. I'm not really sure how else to describe this so any suggestions are welcome! 

I also can't really think of a practical reason why we would want to control these separately. @kokosing mentioned that it would be beneficial to be able to hide column metadata & throw errors on access denied, vs hiding columns completely so that's why I'm leaving it here. 

(Slack link: https://trinodb.slack.com/archives/CP1MUNEUX/p1620418412085600?thread_ts=1618261084.236200&cid=CP1MUNEUX)